experiments/modern/src/runtime/MakiObject.js

```diff
@@ -1,10 +1,22 @@
 const Emitter = require("../Emitter");
 
 class MakiObject {
-  constructor() {
+  constructor(node, parent) {
+    this.xmlNode = node;
+    this.parent = parent;
+    this.children = [];
+    this.hooks = {};
     this._emitter = new Emitter();
   }
 
+  addChild(child) {
+    this.children.push(child);
+  }
+
+  addChildren(children) {
+    this.children = this.children.concat(children);
+  }
```

Comment:
I'd love to keep a convention of all methods that are not found in `std.mi` are prefixed with `js_`